pkg/decision/rollout_service.go

```diff
@@ -30,25 +30,61 @@ import (
 type RolloutService struct {
 	audienceTreeEvaluator     evaluator.TreeEvaluator
 	experimentBucketerService ExperimentService
-	logger logging.OptimizelyLogProducer
+	logger                    logging.OptimizelyLogProducer
 }
 
 // NewRolloutService returns a new instance of the Rollout service
 func NewRolloutService(sdkKey string) *RolloutService {
 	return &RolloutService{
-		logger:logging.GetLogger(sdkKey, "RolloutService"),
+		logger:                    logging.GetLogger(sdkKey, "RolloutService"),
 		audienceTreeEvaluator:     evaluator.NewMixedTreeEvaluator(),
 		experimentBucketerService: NewExperimentBucketerService(logging.GetLogger(sdkKey, "ExperimentBucketerService")),
 	}
 }
 
 // GetDecision returns a decision for the given feature and user context
 func (r RolloutService) GetDecision(decisionContext FeatureDecisionContext, userContext entities.UserContext) (FeatureDecision, error) {
+
 	featureDecision := FeatureDecision{
 		Source: Rollout,
 	}
 	feature := decisionContext.Feature
 	rollout := feature.Rollout
+
+	evaluateConditionTree := func(experiment entities.Experiment) bool {
```

Comment:
since this is a help function, can you try passing by reference instead of value?